### PR TITLE
parse junit file

### PR DESCRIPTION
**Pull request by nadvornik**
depends on https://github.com/os-autoinst/openQA/pull/384 and https://github.com/os-autoinst/openQA/pull/385

### CODE REVIEW

File: testapi.pm [hunk 1]
```diff
@@ -704,6 +705,85 @@ sub console($) {
     }
 }
 
+=head2 parse_junit_log
+
+parse_junit_log("report.xml")
+
+Parse jUnit log file and creates extra test result from it. The file must
+be first uploaded by upload_logs().
+
```
Comment (coolo) [inline testapi.pm]: what's the point here? Why upload and parse in one go?

Comment (coolo) [inline testapi.pm]: it's not like we're going to parse unuploaded files

File: testapi.pm [hunk 2]
```diff
@@ -704,6 +705,85 @@ sub console($) {
...
+        my $ts_category = $ts->{package};
+        $ts_category =~ s/[^A-Za-z0-9._-]/_/g;    # the name is used as part of url so we must strip special characters
+        my $ts_name = $ts_category;
+        $ts_category =~ s/\..*$//;
+        $ts_name =~ s/^[^.]*\.//;
+        $ts_name =~ s/\./_/;
+
+        push @tests,
+          {
+            "flags"    => {"important" => 1},
```
Comment (coolo) [inline testapi.pm]: you guys have some weird relationship to quotes :)

But as a matter of fact none of the quotes is required here.

File: testapi.pm [hunk 3]
```diff
@@ -704,6 +705,85 @@ sub console($) {
...
+        my $ts_result = 'ok';
+        $ts_result = 'fail' if $ts->{failures} || $ts->{errors};
+
+        my $result = {
+            "result"  => $ts_result,
+            "details" => [],
+            "dents"   => 0,
+        };
+
+        my $i = 1;
```
Comment (coolo) [inline testapi.pm]: It's not obvious what $i is - please pick a better name. It only becomes clear that it's a counter 28 lines later

File: testapi.pm [hunk 4]
```diff
@@ -704,6 +705,85 @@ sub console($) {
...
+                }
+                close $fd;
+                $details->{text} = $text_fn;
+            }
+            push @{$result->{details}}, $details;
+            $i++;
+        }
+
+        my $fn = bmwqemu::result_dir() . sprintf("/result-%s.json", $ts_name);
+        bmwqemu::save_json_file($result, $fn);
```
Comment (coolo) [inline testapi.pm]: why use sprintf here? "/result-$ts_name.json" is quite readable IMO. And it's not as if we wouldn't use this form everywhere else.

**Comment by nadvornik** [comment]
upload_logs is the easiest way to transfer the file from VM to os-autoinst where it is parsed.
Keeping the original jUnit file is IMHO a good idea anyway.

Other issues are fixed.

**Comment by coolo** [comment]
Sorry, I mistyped. What I wanted to ask: why *not* upload and parse in one call. I don't see a use case of parsing *without* uploading too

**Comment by coolo** [comment]
please rebase - and answer :)

**Comment by nadvornik** [comment]
ok, uploading is now part of the function too.

File: testapi.pm [hunk 5]
```diff
@@ -720,6 +721,84 @@ sub assert_shutdown(;$) {
...
+file is then parsed as jUnit format and extra test results are created from it.
+
+=cut
+
+sub parse_junit_log {
+    my ($file) = @_;
+
+    upload_logs($file);
+
+    $file =~ s/.*\///;
```
Comment (coolo) [inline testapi.pm]: we use basename in this file already - and you should too because it makes quite more obvious what's going on.

**Pull request merged**